src/main/java/org/jenkinsci/plugins/github_branch_source/PushGHEventSubscriber.java

```diff
@@ -308,6 +322,39 @@ public String getSourceName() {
                             new AbstractGitSCMSource.SCMRevisionImpl(head, push.getHead()));
                 }
             }
+            if (context.wantTags() && ref.startsWith(R_TAGS)) {
+                // NOTE: GitHub provides the timestamp of the head commit, but if this is an annotated tag
+                // then that would be an incorrect timestamp, so we have to assume we are going to have the
+                // wrong timestamp for everything except lightweight tags.
+                //
+                // Now in any case, this actually does not matter.
+                //
+                // Event consumers are supposed to *not* trust the details reported by an event, it's just a hint.
+                // All we really want is that we report enough of a head to provide the head.getName()
+                // then the event consumer is supposed to turn arround and do a fetch(..., event, ...)
+                // and as GitHubSCMSourceRequest strips out the timestamp in calculating the requested
+                // tag names, we have a winner.
+                //
+                // So let's make the assumption that tags are not pushed a long time after their creation and
+                // use the event timestamp. This may cause issues if anyone has a pre-filter that filters
+                // out tags that are less than X seconds old, but as such a filter would be incompatible with events
+                // discovering tags, no harm... the key part is that a pre-filter that removes tags older than X days
+                // will not strip the tag *here* (because it will always be only a few seconds "old"), but when
+                // the fetch call actually has the real tag date the pre-filter will apply at that point in time.
```

Comment:
Uh, OK. I guess this is all hypothetical unless and until someone writes such a filter.